### ISSUE TITLE
Cannot grant user access to a connection in nested collection groups

### ISSUE THREAD

**Issue by fastlorenzo** [scicore-unibas-ch/ansible-modules-guacamole#23]
Hi there,

First of all, thanks a lot for this awesome plugin! Wish I've found it before implementing everything in pure Ansible 😄 

Now that I'm trying to use it, I've noticed a small limitation: you cannot grant user access to a connection which is not in the ROOT collection group.

I believe the issue is related to the piece of code below, which doesn't work when you have 2 levels of nesting: 
https://github.com/scicore-unibas-ch/ansible-modules-guacamole/blob/18d44a6560895610a015665789a6f54b086b4c00/plugins/module_utils/guacamole.py#L61-L91

For some context, below is my structure for connections/collection groups that doesn't work:
![image](https://user-images.githubusercontent.com/768639/169639774-c26b3b17-ed8d-488f-a76f-98b73dc64ba8.png)

I did additional testing with placing the connections directly under `students` (which is under ROOT), and that works:
![image](https://user-images.githubusercontent.com/768639/169640317-c8f4afd4-b83b-4b18-94f7-61cef6144ed3.png)

Below is the result of the API call used to retrieve the tree of connections (groups) under ROOT, (with 2 levels of nesting, which doesn't work):
```
curl -s https://DOMAIN/api/session/data/postgresql/connectionGroups/ROOT/tree?token=XYZ | jq
{
  "name": "ROOT",
  "identifier": "ROOT",
  "type": "ORGANIZATIONAL",
  "activeConnections": 0,
  "childConnectionGroups": [
    {
      "name": "students",
      "identifier": "21",
      "parentIdentifier": "ROOT",
      "type": "ORGANIZATIONAL",
      "activeConnections": 0,
      "childConnectionGroups": [
        {
          "name": "vmkstu02",
          "identifier": "22",
          "parentIdentifier": "21",
          "type": "ORGANIZATIONAL",
          "activeConnections": 0,
          "childConnections": [
            {
              "name": "vmkstu02 - Kali - SSH",
              "identifier": "23",
              "parentIdentifier": "22",
              "protocol": "ssh",
              "attributes": {
                "guacd-encryption": null,
                "failover-only": null,
                "weight": null,
                "max-connections": "1",
                "guacd-hostname": null,
                "guacd-port": null,
                "max-connections-per-user": null
              },
              "activeConnections": 0
            },
            {
              "name": "vmkstu02 - Kali - RDP",
              "identifier": "24",
              "parentIdentifier": "22",
              "protocol": "rdp",
              "attributes": {
                "guacd-encryption": null,
                "failover-only": null,
                "weight": null,
                "max-connections": "1",
                "guacd-hostname": null,
                "guacd-port": null,
                "max-connections-per-user": null
              },
              "activeConnections": 0
            }
          ],
          "attributes": {
            "max-connections": null,
            "max-connections-per-user": null,
            "enable-session-affinity": ""
          }
        },
        {
          "name": "vmkstu01",
          "identifier": "23",
          "parentIdentifier": "21",
          "type": "ORGANIZATIONAL",
          "activeConnections": 0,
          "childConnections": [
            {
              "name": "vmkstu01 - Kali - RDP",
              "identifier": "26",
              "parentIdentifier": "23",
              "protocol": "rdp",
              "attributes": {
                "guacd-encryption": null,
                "failover-only": null,
                "weight": null,
                "max-connections": "1",
                "guacd-hostname": null,
                "guacd-port": null,
                "max-connections-per-user": null
              },
              "activeConnections": 0
            },
            {
              "name": "vmkstu01 - Kali - SSH",
              "identifier": "25",
              "parentIdentifier": "23",
              "protocol": "ssh",
              "attributes": {
                "guacd-encryption": null,
                "failover-only": null,
                "weight": null,
                "max-connections": "1",
                "guacd-hostname": null,
                "guacd-port": null,
                "max-connections-per-user": null
              },
              "activeConnections": 0
            }
          ],
          "attributes": {
            "max-connections": null,
            "max-connections-per-user": null,
            "enable-session-affinity": ""
          }
        }
      ],
      "attributes": {
        "max-connections": null,
        "max-connections-per-user": null,
        "enable-session-affinity": ""
      }
    },
    {
      "name": "services",
      "identifier": "16",
      "parentIdentifier": "ROOT",
      "type": "ORGANIZATIONAL",
      "activeConnections": 0,
      "attributes": {
        "max-connections": null,
        "max-connections-per-user": null,
        "enable-session-affinity": ""
      }
    },
    {
      "name": "lab",
      "identifier": "17",
      "parentIdentifier": "ROOT",
      "type": "ORGANIZATIONAL",
      "activeConnections": 0,
      "attributes": {
        "max-connections": null,
        "max-connections-per-user": null,
        "enable-session-affinity": ""
      }
    },
    {
      "name": "course",
      "identifier": "18",
      "parentIdentifier": "ROOT",
      "type": "ORGANIZATIONAL",
      "activeConnections": 0,
      "attributes": {
        "max-connections": null,
        "max-connections-per-user": null,
        "enable-session-affinity": ""
      }
    }
  ],
  "attributes": {}
}
```

**Comment by fastlorenzo** [comment]
I think this would require a change here as well to support multiple levels of nesting: 
https://github.com/scicore-unibas-ch/ansible-modules-guacamole/blob/18d44a6560895610a015665789a6f54b086b4c00/plugins/modules/guacamole_user.py#L544-L550

The entire tree will need to be traversed until ROOT is reached in order to grant access to all necessary parent connection groups.

Proposed approach is to use recursive calls and create a specific function to build the list of connections / connection groups to be allowed.

**Comment by pescobar** [comment]
Solved in https://github.com/scicore-unibas-ch/ansible-modules-guacamole/pull/40 

Published [new release 0.0.23](https://github.com/scicore-unibas-ch/ansible-modules-guacamole/releases/tag/0.0.23) with the fix
